### PR TITLE
IGNITE-17687 Get rid of using deprecated exception constructors in the sql-engine module.

### PR DESCRIPTION
**Pull request by xtern**

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java [hunk 1]
```diff
@@ -333,9 +334,9 @@ public static <T> T await(CompletionStage<T> stage) {
         try {
             return stage.toCompletableFuture().get();
         } catch (ExecutionException e) {
-            throw new IgniteException(e.getCause());
+            throw new IgniteException(OPERATION_INTERRUPTED_ERR, e.getCause());
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java:337]: I think we should use SqlException here and below

Comment (xtern) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/api/SessionImpl.java]: Done, thanks

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 2]
```diff
@@ -344,7 +347,8 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
         SchemaPlus schema = sqlSchemaManager.schema(schemaName);
 
         if (schema == null) {
-            return CompletableFuture.failedFuture(new IgniteInternalException(format("Schema not found [schemaName={}]", schemaName)));
+            return CompletableFuture.failedFuture(
+                    new IgniteInternalException(SCHEMA_NOT_FOUND_ERR, format("Schema not found [schemaName={}]", schemaName)));
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:351]: I think it should be public exception

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 3]
```diff
@@ -361,7 +365,7 @@ private CompletableFuture<AsyncSqlCursor<List<Object>>> querySingle0(
         try {
             session.registerResource(closeableResource);
         } catch (IllegalStateException ex) {
-            return CompletableFuture.failedFuture(new IgniteInternalException(
+            return CompletableFuture.failedFuture(new IgniteInternalException(SESSION_EXPIRED_ERR,
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:368]: It looks also as SqlException

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java [hunk 4]
```diff
@@ -449,7 +453,7 @@ private List<CompletableFuture<AsyncSqlCursor<List<Object>>>> query0(
         SchemaPlus schema = sqlSchemaManager.schema(schemaName);
 
         if (schema == null) {
-            throw new IgniteInternalException(format("Schema not found [schemaName={}]", schemaName));
+            throw new IgniteInternalException(SCHEMA_NOT_FOUND_ERR, format("Schema not found [schemaName={}]", schemaName));
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java:456]: the same

**Pull request merged**